### PR TITLE
FW/Win32: add a crash context dumper for Windows

### LINKED ISSUE #314: Add support for context-dumping-on-crash on Windows
The [AddVectoredExceptionHandler()](https://learn.microsoft.com/en-us/windows/win32/api/errhandlingapi/nf-errhandlingapi-addvectoredexceptionhandler) function does give us the context.
```c++
#include <windows.h>

static LONG WINAPI handler(EXCEPTION_POINTERS *info)
{
    return EXCEPTION_CONTINUE_SEARCH;
}

int main()
{
    asm("vpcmpeqb %ymm15, %ymm15, %ymm15");
    PVOID h = AddVectoredExceptionHandler(true, handler);
    __builtin_trap();
}
```
In the debugger:
```
(lldb) Process 9736 stopped
* thread #1, stop reason = breakpoint 1.1
    frame #0: 0x00007ff652af1458 test.exe`handler(info=0x0000000ba91fef50)
(lldb) p/x *info->ContextRecord
(_CONTEXT) $2 = {
  P1Home = 0x0000000000000400
  P2Home = 0x000002635493151a
  P3Home = 0x0000000000000008
  P4Home = 0x0000026354798b90
  P5Home = 0x0000000000000004
  P6Home = 0x0000000000000018
  ContextFlags = 0x0010005f
  MxCsr = 0x00001f80
  SegCs = 0x0033
  SegDs = 0x002b
  SegEs = 0x002b
  SegFs = 0x0053
  SegGs = 0x002b
  SegSs = 0x002b
  EFlags = 0x00010202
  Dr0 = 0x0000000000000000
  Dr1 = 0x0000000000000000
  Dr2 = 0x0000000000000000
  Dr3 = 0x0000000000000000
  Dr6 = 0x0000000000000000
  Dr7 = 0x0000000000000000
  Rax = 0x0000026354798b40
  Rcx = 0x00007ff8f58d4008
  Rdx = 0x0000000000000000
  Rbx = 0x0000000000000008
  Rsp = 0x000000bd35bffca0
  Rbp = 0x000000bd35bffcd0
  Rsi = 0x0000000000000011
  Rdi = 0x00000263549314a8
  R8 = 0x000000007ffe4000
  R9 = 0x0000000000000083
  R10 = 0x0000000000000001
  R11 = 0xfffffffff6cd6977
  R12 = 0x00000263549314e0
  R13 = 0x0000000000000000
  R14 = 0x0000000000000000
  R15 = 0x0000000000000000
  Rip = 0x00007ff6f120148d
   = {
    FltSave = {
      ControlWord = 0x037f
      StatusWord = 0x0000
      TagWord = 0x00
      Reserved1 = 0x00
      ErrorOpcode = 0x0000
      ErrorOffset = 0x00000000
      ErrorSelector = 0x0000
      Reserved2 = 0x0000
      DataOffset = 0x00000000
      DataSelector = 0x0000
      Reserved3 = 0x0000
      MxCsr = 0x00001f80
      MxCsr_Mask = 0x0000ffff
      FloatRegisters = {
        [0] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [1] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [2] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [3] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [4] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [5] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [6] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [7] = (Low = 0x0000000000000000, High = 0x0000000000000000)
      }
      XmmRegisters = {
        [0] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [1] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [2] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [3] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [4] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [5] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [6] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [7] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [8] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [9] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [10] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [11] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [12] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [13] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [14] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [15] = (Low = 0xffffffffffffffff, High = 0xffffffffffffffff)
      }
      Reserved4 = "\0\0yTc\U00000002\0\0\0\0\0\0\0\0\0\0\0\0yTc\U00000002\0\0\x83?\x85\xf5\xf8\U0000007f\0\0\xa0\x8byTc\U00000002\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\U00000001\U00000014\x93Tc\U00000002\0\0\xa0\x8byTc\U00000002\0\0\U00000011\0\0\0\0\0\0\0p"
    }
    FloatSave = {
      ControlWord = 0x037f
      StatusWord = 0x0000
      TagWord = 0x00
      Reserved1 = 0x00
      ErrorOpcode = 0x0000
      ErrorOffset = 0x00000000
      ErrorSelector = 0x0000
      Reserved2 = 0x0000
      DataOffset = 0x00000000
      DataSelector = 0x0000
      Reserved3 = 0x0000
      MxCsr = 0x00001f80
      MxCsr_Mask = 0x0000ffff
      FloatRegisters = {
        [0] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [1] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [2] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [3] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [4] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [5] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [6] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [7] = (Low = 0x0000000000000000, High = 0x0000000000000000)
      }
      XmmRegisters = {
        [0] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [1] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [2] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [3] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [4] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [5] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [6] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [7] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [8] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [9] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [10] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [11] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [12] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [13] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [14] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [15] = (Low = 0xffffffffffffffff, High = 0xffffffffffffffff)
      }
      Reserved4 = "\0\0yTc\U00000002\0\0\0\0\0\0\0\0\0\0\0\0yTc\U00000002\0\0\x83?\x85\xf5\xf8\U0000007f\0\0\xa0\x8byTc\U00000002\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\U00000001\U00000014\x93Tc\U00000002\0\0\xa0\x8byTc\U00000002\0\0\U00000011\0\0\0\0\0\0\0p"
    }
     = {
      Header = {
        [0] = (Low = 0x000000000000037f, High = 0x0000000000000000)
        [1] = (Low = 0x0000000000000000, High = 0x0000ffff00001f80)
      }
      Legacy = {
        [0] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [1] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [2] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [3] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [4] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [5] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [6] = (Low = 0x0000000000000000, High = 0x0000000000000000)
        [7] = (Low = 0x0000000000000000, High = 0x0000000000000000)
      }
      Xmm0 = (Low = 0x0000000000000000, High = 0x0000000000000000)
      Xmm1 = (Low = 0x0000000000000000, High = 0x0000000000000000)
      Xmm2 = (Low = 0x0000000000000000, High = 0x0000000000000000)
      Xmm3 = (Low = 0x0000000000000000, High = 0x0000000000000000)
      Xmm4 = (Low = 0x0000000000000000, High = 0x0000000000000000)
      Xmm5 = (Low = 0x0000000000000000, High = 0x0000000000000000)
      Xmm6 = (Low = 0x0000000000000000, High = 0x0000000000000000)
      Xmm7 = (Low = 0x0000000000000000, High = 0x0000000000000000)
      Xmm8 = (Low = 0x0000000000000000, High = 0x0000000000000000)
      Xmm9 = (Low = 0x0000000000000000, High = 0x0000000000000000)
      Xmm10 = (Low = 0x0000000000000000, High = 0x0000000000000000)
      Xmm11 = (Low = 0x0000000000000000, High = 0x0000000000000000)
      Xmm12 = (Low = 0x0000000000000000, High = 0x0000000000000000)
      Xmm13 = (Low = 0x0000000000000000, High = 0x0000000000000000)
      Xmm14 = (Low = 0x0000000000000000, High = 0x0000000000000000)
      Xmm15 = (Low = 0xffffffffffffffff, High = 0xffffffffffffffff)
    }
  }
  VectorRegister = {
    [0] = (Low = 0x0000026354798b90, High = 0x0000000000000040)
    [1] = (Low = 0x0000026354798bc0, High = 0x0000000000000000)
    [2] = (Low = 0x0000026354930000, High = 0x00007ff8f5853f83)
    [3] = (Low = 0x0000000000000008, High = 0x0000000040000060)
    [4] = (Low = 0x0000000040000062, High = 0x0000000000000001)
    [5] = (Low = 0x0000000000000000, High = 0x00007ff8f580d1f0)
    [6] = (Low = 0x0000026354790000, High = 0x0000000050000163)
    [7] = (Low = 0x0000026354790000, High = 0x0000000000000000)
    [8] = (Low = 0x0000026354931550, High = 0x0000000000000000)
    [9] = (Low = 0x0000000000000000, High = 0x0000000000000000)
    [10] = (Low = 0x0000000000000011, High = 0x0000000000000000)
    [11] = (Low = 0x0000000000000000, High = 0x0000000000000002)
    [12] = (Low = 0x0000000000000000, High = 0x00007ff8f580d1f0)
    [13] = (Low = 0x0000026300000000, High = 0x0000000050000163)
    [14] = (Low = 0x0000026354930000, High = 0x00007ff8f5780000)
    [15] = (Low = 0x0000026354790000, High = 0x00007ff8402c0062)
    [16] = (Low = 0x0000000000000400, High = 0x0000000000000480)
    [17] = (Low = 0x0000000000000000, High = 0x0000000000000000)
    [18] = (Low = 0x0000000000000000, High = 0x000000bd35bff9b0)
    [19] = (Low = 0x0000000000000030, High = 0x0000000000000000)
    [20] = (Low = 0x0000000000000000, High = 0x0000000000000000)
    [21] = (Low = 0x0000000000000000, High = 0x0000000000000000)
    [22] = (Low = 0x000002635476c000, High = 0x000002635476816d)
    [23] = (Low = 0x000000bd35bffb98, High = 0x0000000000000000)
    [24] = (Low = 0x00007ff6f1201000, High = 0x00007ff6f1200000)
    [25] = (Low = 0x0000000000000009, High = 0x0000f5f0ffc8beee)
  }
  VectorControl = 0x0000000000000000
  DebugControl = 0x00000263547637a0
  LastBranchToRip = 0x0000000000000000
  LastBranchFromRip = 0x0000000000000000
  LastExceptionToRip = 0x0000000000000000
  LastExceptionFromRip = 0x0000000000000000
}
```

The `info.ContextRecord->ContextFlags`  value (0x10005f) indicates that this is using XSAVE, so the rest of the state is there.  

Calling [LocateXStateFeature](https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-locatexstatefeature) on that context state does return a pointer that appears to be valid.
```
(gdb) p &info->ContextRecord.FloatSave.XmmRegisters
$1 = (M128A (*)[16]) 0x5ff8e0
(gdb) p ((void *(*)(PCONTEXT, DWORD, PDWORD))LocateXStateFeature)(info->ContextRecord, 1, 0)
$2 = (void *) 0x5ff8e0
(gdb) p ((void *(*)(PCONTEXT, DWORD, PDWORD))LocateXStateFeature)(info->ContextRecord, 2, 0)
$3 = (void *) 0x5ffd40
(gdb) p/x info->ContextRecord.FloatSave.XmmRegisters[15]
$6 = {Low = 0xffffffffffffffff, High = 0xffffffffffffffff}
(gdb) x/2gx $3 + 15*16
0x5ffe30:       0xffffffffffffffff      0xffffffffffffffff
```

See
- https://learn.microsoft.com/en-us/windows/win32/debug/avx-support-portal
- https://learn.microsoft.com/en-us/windows/win32/debug/working-with-xstate-context

### PR DESCRIPTION
**Pull request by thiagomacieira**
Fixes #314.

Now we're getting:
```yaml
   - level: info
      text: |1
       Registers:
        rax   = 0x000000000000002a (42)
        rbx   = 0x0000000000000000 (0)
        rcx   = 0x000000000000feed
        rdx   = 0x0000000000c0ffee
        rsi   = 0x000002a23c9873e0
        rdi   = 0x00000000dfef324d
        rbp   = 0x000000e6ed5ffc10
        rsp   = 0x000000e6ed5ffb90
        r8    = 0x00000000dfef324d
        r9    = 0x000002a23c9873e0
        r10   = 0x0000000000000011 (17)
        r11   = 0x000000e6ed5ffcc0
        r12   = 0x0000000000000000 (0)
        r13   = 0x0000000000000000 (0)
        r14   = 0x0000000000000000 (0)
        r15   = 0x0000000000000000 (0)
        rip   = 0x00007ff6043e9896
        flags = 0x00010246 [ PF ZF IF RF ]
        fs    = 0x53
        gs    = 0x2b
        fcw   = 0x37f
        fsw   = 0x3000 [ top=6 ]
        ftw   = 0xc0
        st(0) = 4000c90fdaa22168c235 (0xc.90fdaa22168c235p-2)
        st(1) = 3fff8000000000000000 (0x8p-3)
        st(2) = 00000000000000000000 (0x0p+0)
        st(3) = 00000000000000000000 (0x0p+0)
        st(4) = 00000000000000000000 (0x0p+0)
        st(5) = 00000000000000000000 (0x0p+0)
        st(6) = 00000000000000000000 (0x0p+0)
        st(7) = 00000000000000000000 (0x0p+0)
        mxcsr = 0x00001f80 [ IM DM ZM OM UM PM RC=nearest ]
        ymm0  = 0000000000000000:0000000000000000 3ff0000000000000:3cb0000000000000
        ymm1  = ffffffffffffffff:ffffffffffffffff ffffffffffffffff:ffffffffffffffff
        ymm2  = 0000000000000000:0000000000000000 7ff8000000000000:fff0000000000000
        ymm3  = 0000000000000000:0000000000000000 00c0ffeedeadbeef:0000feed0000002a
        ymm4  = 0000000000000000:0000000000000000 3f80000000000000:bfc000007f800000
        ymm5  = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
        ymm6  = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
        ymm7  = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
        ymm8  = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
        ymm9  = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
        ymm10 = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
        ymm11 = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
        ymm12 = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
        ymm13 = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
        ymm14 = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
        ymm15 = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
```

which matches expectations:
* `rax` contains 42 (from `"a" (42)`)
* `rcx` contains 0xfeed (from `"c" (0xfeed)`)
* `rdx` contains 0xc0ffee (from `"d" (0xc0ffee)`)
* `st(0)` contains `acosl(-1)` (pi)
* `st(1)` contains 1.0
* MXCSR is 0x1f80
* `ymm0` contains `_mm_set_pd(1, std::numeric_limits<double>::epsilon())`
* `ymm1` contains `_mm_set1_epi32(-1)` extended to all 256 bits
* `ymm2` contains `_mm_setpd(quiet_NaN, -infinity)`
* `ymm3` contains `_mm_setr_epi32(42, 0xfeed, 0xdeadbeef, 0xc0ffee)`
* `ymm4` contains `_mm_set_ps(1, 0, -1.5, infinity)`

### CODE REVIEW

File: framework/sysdeps/windows/child_debug.cpp [hunk 1]
```diff
@@ -5,15 +5,59 @@
...
 
 void debug_init_global(const char *on_hang_arg, const char *on_crash_arg)
 {
     (void) on_hang_arg;
-    (void) on_crash_arg;
+    if (SandstoneConfig::ChildDebugCrashes) {
+        std::string_view arg = on_crash_arg;
+        if (arg == "context") {
+            sApp->shmem->debug_mailslot = open_mailslot(MemfdInheritOnExec);
+        } else if (arg != "kill") {
```
Comment (lukasz-tuz) [inline framework/sysdeps/windows/child_debug.cpp:56]: We're going to need to either update bats along with this change or handle other `on-crash` actions here.

**Review by lukasz-tuz — CHANGES_REQUESTED** [review]

File: framework/sysdeps/windows/child_debug.cpp (hunk 1 above)
Comment (thiagomacieira) [reply framework/sysdeps/windows/child_debug.cpp:56]: Fixed.

File: framework/sysdeps/windows/win32_errorstrings.h [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    case STATUS_ILLEGAL_INSTRUCTION:
+    case STATUS_PRIVILEGED_INSTRUCTION:
+        return "Illegal instruction";
+    case STATUS_IN_PAGE_ERROR:
+        return "Paging violation";
+    case STATUS_INTEGER_DIVIDE_BY_ZERO:
+        return "Integer division by zero";
+    case STATUS_NO_MEMORY:
+        return "Out of memory condition";
+    case STATUS_STACK_BUFFER_OVERRUN:
```
Comment (thiagomacieira) [inline framework/sysdeps/windows/win32_errorstrings.h:50]: Though https://devblogs.microsoft.com/oldnewthing/20230731-00/?p=108505 ("Misinterpreting the misleadingly-named STATUS_STACK_BUFFER_OVERRUN"), https://devblogs.microsoft.com/oldnewthing/20190108-00/?p=100655 ("STATUS_STACK_BUFFER_OVERRUN doesn’t mean that there was a stack buffer overrun")

**Comment by thiagomacieira** [comment]
Testing indicates a multi-children crash blames the wrong thread:
```yaml
  test-plans:
    fullsocket: [ { starting_cpu: 0, count: 2 }, { starting_cpu: 2, count: 2 } ]
    heuristic: [ { starting_cpu: 0, count: 1 }, { starting_cpu: 1, count: 1 }, { starting_cpu: 2, count: 1 }, { starting_cpu: 3, count: 1 } ]
  tests:
  - test: selftest_sigsegv_socket1
    details: { quality: beta, description: "Crashes with SIGSEGV (data) dereferencing the null page for any thread of socket 1" }
    state: { seed: 'LCG:1100863826', iteration: 0, retry: false }
    time-at-start: { elapsed:      0.000, now: !!timestamp '2023-10-19T23:32:57Z' }
    result: crash
    result-details: { crashed: true, core-dump: false, code: 0xc0000005, reason: 'Access violation' }
    fail: { cpu-mask: 'X.:..', time-to-fail: null, seed: 'LCG:1100863826'}
    time-at-end:   { elapsed:     54.479, now: !!timestamp '2023-10-19T23:32:57Z' }
    test-runtime: 54.361
    threads:
    - thread: 0
      id: { logical-group:  0, logical:  0, package: 0, core: 0, thread: 0, family: 6, model: 0x3c, stepping: 0, microcode: null, ppin: null }
      state: failed
      time-to-fail: 52.508
      loop-count: 0
      messages:
      - { level: error, text: 'E> Received exception 0xc0000005 (Access violation), RIP = 0x7ff6464699bc (base+0x299bc), CR2 = 0x645 access=R' }
      - level: info
        text: |1
         Registers:
          rax   = 0x0000000000000645 (1605)
          rbx   = 0x0000000000000000 (0)
          rcx   = 0x0000000000000645 (1605)
          rdx   = 0x0000014d2a6f6e80
          rsi   = 0x0000014d2a6f2050
          rdi   = 0x0000000000000000 (0)
          rbp   = 0x0000002d00fff620
          rsp   = 0x0000002d00fff620
          r8    = 0x0000002d00fff5c8
          r9    = 0x0000000000000000 (0)
          r10   = 0x0000000000000000 (0)
          r11   = 0x0000002d00fff670
          r12   = 0x0000000000000000 (0)
          r13   = 0x0000000000000000 (0)
          r14   = 0x0000000000000000 (0)
          r15   = 0x0000000000000000 (0)
          rip   = 0x00007ff6464699bc
          flags = 0x00010202 [ IF RF ]
          fs    = 0x53
          gs    = 0x2b
          fcw   = 0x37f
          fsw   = 0 [ top=0 ]
          ftw   = 0
          mxcsr = 0x00001f80 [ IM DM ZM OM UM PM RC=nearest ]
          ymm0  = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
          ymm1  = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
          ymm2  = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
          ymm3  = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
          ymm4  = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
          ymm5  = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
          ymm6  = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
          ymm7  = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
          ymm8  = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
          ymm9  = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
          ymm10 = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
          ymm11 = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
          ymm12 = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
          ymm13 = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
          ymm14 = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
          ymm15 = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
      - { level: error, text: 'E> Received exception 0xc0000005 (Access violation), RIP = 0x7ff6464699bc (base+0x299bc), CR2 = 0xdc2 access=R' }
      - level: info
        text: |1
         Registers:
          rax   = 0x0000000000000dc2 (3522)
          rbx   = 0x0000000000000000 (0)
          rcx   = 0x0000000000000dc2 (3522)
          rdx   = 0x000001dbc5f96ec0
          rsi   = 0x000001dbc5f92060
          rdi   = 0x0000000000000000 (0)
          rbp   = 0x000000488a3ff8b0
          rsp   = 0x000000488a3ff8b0
          r8    = 0x000000488a3ff858
          r9    = 0x0000000000000000 (0)
          r10   = 0x0000000000000000 (0)
          r11   = 0x000000488a3ff900
          r12   = 0x0000000000000000 (0)
          r13   = 0x0000000000000000 (0)
          r14   = 0x0000000000000000 (0)
          r15   = 0x0000000000000000 (0)
          rip   = 0x00007ff6464699bc
          flags = 0x00010202 [ IF RF ]
          fs    = 0x53
          gs    = 0x2b
          fcw   = 0x37f
          fsw   = 0 [ top=0 ]
          ftw   = 0
          mxcsr = 0x00001f80 [ IM DM ZM OM UM PM RC=nearest ]
          ymm0  = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
          ymm1  = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
          ymm2  = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
          ymm3  = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
          ymm4  = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
          ymm5  = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
          ymm6  = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
          ymm7  = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
          ymm8  = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
          ymm9  = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
          ymm10 = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
          ymm11 = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
          ymm12 = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
          ymm13 = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
          ymm14 = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
          ymm15 = 0000000000000000:0000000000000000 0000000000000000:0000000000000000
```

File: framework/sysdeps/windows/win32_errorstrings.h [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        return "Stack buffer overrun";
+    case STATUS_STACK_OVERFLOW:
+        return "Staack overflow";
```
Comment (AthenasJimenez) [inline framework/sysdeps/windows/win32_errorstrings.h:53]: Small typo: "Staack -> Stack"

File: framework/sysdeps/windows/win32_errorstrings.h (hunk 2 above)
Comment (AthenasJimenez) [reply framework/sysdeps/windows/win32_errorstrings.h:50]: Uhm, maybe adding "Stack buffer overrun or abnormal termination"?

File: framework/sysdeps/unix/child_debug.cpp [hunk 4]
```diff
@@ -866,6 +828,7 @@ static bool print_signal_info(const CrashContext::Fixed &ctx)
     if (ctx.rip != ctx.crash_address)
         message += stdprintf(", CR2 = %p", ctx.crash_address);
     if (ctx.trap_nr >= 0) {
+#ifdef __x86_64__
```
Comment (AthenasJimenez) [inline framework/sysdeps/unix/child_debug.cpp:831]: This looks weird to me, shouldn't be better to include the if inside of the "#ifdef __x86_64__"?

File: framework/sysdeps/windows/child_debug.cpp [hunk 5]
```diff
@@ -3,22 +3,306 @@
...
+    if (!SetEvent(HANDLE(sApp->shmem->debug_event)))
+        win32_perror("SetEvent for mailslot");
+
+    TerminateProcess(GetCurrentProcess(), info->ExceptionRecord->ExceptionCode);
+    __builtin_unreachable();
+    return EXCEPTION_CONTINUE_SEARCH;
+}
+
+#ifdef __x86_64__
+static void print_non_gprs(FILE *log, PCONTEXT ctx)
```
Comment (AthenasJimenez) [inline framework/sysdeps/windows/child_debug.cpp:132]: For my self education, what info is going to be printed here?

**Review by AthenasJimenez — APPROVED** [review]

**Pull request merged**